src/PhpBrew/VariantBuilder.php

```diff
@@ -315,34 +315,27 @@ public function __construct()
             return;
         };
 
-        $this->variants['curl'] = function (Build $build, $prefix = null) {
-            if ($prefix) {
-                return "--with-curl=$prefix";
-            }
-
-            if ($prefix = Utils::findIncludePrefix('curl/curl.h')) {
-                return "--with-curl=$prefix";
-            }
+        $this->variants['curl'] = function (Build $build, $val = null) {
+            $prefix = Utils::findPrefix(array(
+                new BrewPrefixFinder('curl'),
+                new PkgConfigPrefixFinder('libcurl'),
+                new IncludePrefixFinder('curl/curl.h'),
+            ));
 
-            if ($prefix = Utils::getPkgConfigPrefix('libcurl')) {
-                return "--with-curl=$prefix";
-            }
+            if ($build->compareVersion('7.4') < 0) {
+                if ($val) {
+                    return "--with-curl=$val";
+                }
 
-            if ($bin = Utils::findBin('brew')) {
-                if ($prefix = exec_line("$bin --prefix curl")) {
-                    if (file_exists($prefix)) {
-                        return "--with-curl=$prefix";
-                    }
-                    echo "homebrew prefix '$prefix' doesn't exist. you forgot to install?\n";
+                if ($prefix !== null) {
+                    return '--with-curl=' . $prefix;
                 }
-            }
-            if ($bin = Utils::findBin('curl-config')) {
-                if ($prefix = exec_line("$bin --prefix")) {
-                    if (file_exists($prefix)) {
-                        return "--with-curl=$prefix";
-                    }
-                    echo "homebrew prefix '$prefix' doesn't exist. you forgot to install?\n";
+            } else {
+                if ($prefix !== null) {
+                    $build->setPkgConfigPath($prefix.'/lib/pkgconfig');
```

Comment:
I'm still struggling to understand the logic being implemented. By the moment when this line of code is reached, I assume the following:

1. We're on macOS.
2. We determined the build prefix (most likely, from Homebrew by using the `BrewPrefixFinder`).
3. Besides passing the `curl` build prefix explicitly to the PHP build scripts, we're passing it the `PKG_CONFIG_PATH` based on this prefix.

The questions are:
1. What is it supposed to use this prefix for?
2. How are prefixes from multiple extensions are supposed to work together?
3. Is this fix specific to these couple of extensions or it's applicable to any Homebrew-based dependencies?
4. What is the specific change in PHP 7.4 that requires this additional logic and why is it needed only for macOS?